docker_mcp/core/docker_context.py

```
@@ -89,32 +89,36 @@ async def _run_docker_command(
         """Ensure Docker context exists for host."""
-        if host_id not in self.config.hosts:
-            raise DockerContextError(f"Host {host_id} not configured")
-
-        # Check cache first
-        if host_id in self._context_cache:
-            context_name = self._context_cache[host_id]
-            if await self._context_exists(context_name):
+        try:
+            async with asyncio.timeout(30.0):  # 30 second timeout for context operations
+                if host_id not in self.config.hosts:
+                    raise DockerContextError(f"Host {host_id} not configured")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)